ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp

```diff
@@ -117,6 +117,12 @@ auto preprocess_inputs(
             Shape repeats(std::array<uint32_t, 4>{first_shape[0], 1, 1, 1});
             second = ttnn::repeat(second, repeats);
         }
+        // repeats second if it is smaller
+        if (first_shape.rank() == 4 and second_shape.rank() == 4 and first_shape[1] > second_shape[1]) {
+            tt::log_warning(tt::LogOp, "Using repeat op to broadcast channel dim");
+            Shape repeats(std::array<uint32_t, 4>{1, first_shape[1], 1, 1});
+            second = ttnn::repeat(second, repeats);
+        }
     };
```

Comment:
Can we not put this temporary fix here instead?  https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp#L110